gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/InstructorGradeSummaryGradesPanel.java

```diff
@@ -107,22 +112,55 @@ public void onBeforeRender() {
 		// setup
 		final List<String> categoryNames = new ArrayList<>();
 		final Map<String, List<Assignment>> categoryNamesToAssignments = new HashMap<>();
+		final Map<Integer, String> sortableCategoryNames = new HashMap<>();
 
 		// iterate over assignments and build map of categoryname to list of assignments
 		for (final Assignment assignment : assignments) {
 
 			final String categoryName = getCategoryName(assignment);
 
 			if (!categoryNamesToAssignments.containsKey(categoryName)) {
-				categoryNames.add(categoryName);
+				// If the category has a sort order add it to the sortable map.
+				// If not, just add the category to the list directly.
+				if (assignment.getCategoryOrder() != null) {
+					sortableCategoryNames.put(assignment.getCategoryOrder(), categoryName);
+				} else {
+					categoryNames.add(categoryName);
+				}
 				categoryNamesToAssignments.put(categoryName, new ArrayList<>());
 			}
 
 			categoryNamesToAssignments.get(categoryName).add(assignment);
 		}
 		Map<String, CategoryDefinition> categoriesMap = businessService.getGradebookCategories().stream()
 				.collect(Collectors.toMap(cat -> cat.getName(), cat -> cat));
-		Collections.sort(categoryNames);
+
+		//If categories are being used, sort various maps and lists by their proper orders
+		if(this.categoriesEnabled){
+			//sort assignments in each category
+			categoryNamesToAssignments.forEach((categoryName, assignmentsInCategory) -> {
+				Collections.sort(assignmentsInCategory);
+			});
+
+			//sort categories, if needed
+			List<Integer> sortedCategoryKeys = new ArrayList<>(sortableCategoryNames.keySet());
+			//null safe integer sort forcing uncategorized to bottom of list
+			Collections.sort(sortedCategoryKeys, (order1, order2) -> {
```

Comment:
I think you can just use nullsFirst for this? https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#nullsFirst-java.util.Comparator-